### PR TITLE
Fix adding sample sheet to housekeeper

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR fixes the issue where the path to the sample sheet in the flow cell run directory was assumed to be in the root when its actual location seems to vary. Instead of hard coding the path, a function has been added which recursively searches the flow cell run directory for it.

### Fixed
- Find sample sheet path in flow cell run directory when transferring housekeeper files in new demultiplex finish

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/meta/demultiplex/demux_post_processing.py [hunk 1]
```diff
@@ -230,18 +231,44 @@ def add_sample_fastq_files(self, flow_cell_directory: Path, flow_cell_name: str)
...
+
+        Args:
+            flow_cell_directory (Path): The path to the flow cell directory.
+
+        Returns:
+            Path: The path to the found sample sheet file.
+
+        Raises:
+            FileNotFoundError: If the sample sheet file is not found in the flow cell directory.
+        """
```
Comment (ChrOertlin) [inline cg/meta/demultiplex/demux_post_processing.py:263]:
```suggestion
        """
        Recursively searches for the given sample sheet file in the provided flow cell directory.
        """
```

**Review by ChrOertlin — APPROVED** [review]
Added a sample sheet file path finder to the demux post processing api and changed function parameter variables.

Looks good, just one docstring that does not follow current convention.

File: cg/meta/demultiplex/demux_post_processing.py (hunk 1 above)
Comment (seallard) [reply cg/meta/demultiplex/demux_post_processing.py:263]: I don't think we have a guidelines for our doc strings yet, I created a suggestion for it here https://github.com/Clinical-Genomics/atlas/pull/1772

Comment (ChrOertlin) [reply cg/meta/demultiplex/demux_post_processing.py:265]: Nice! Curretnly I think we use single sentence Docstrings, at least that is how it was communicated before by @henrikstranneheim if I am not mistaken. Maybe it is better to stick to that until something is decided.

Comment (seallard) [reply cg/meta/demultiplex/demux_post_processing.py:265]: Ok! I'll keep the exception part though since I think it is valuable

**Comment by seallard** [comment]
Tested in stage

**Pull request merged**